packages/office-ui-fabric-react/src/components/Image/Image.tsx

```diff
@@ -124,21 +115,49 @@ export class Image extends BaseComponent<IImageProps, IImageState> {
     );
   }
 
-  private _evaluateImage(): boolean {
+  @autobind
+  private _onImageLoaded(ev: React.SyntheticEvent<HTMLImageElement>): void {
+    let { src, onLoad } = this.props;
+    if (onLoad) {
+      onLoad(ev);
+    }
+
+    this._computeCoverStyle(this.props);
+
+    if (src) {
+      this._setStateToLoadedOrErrorLoaded();
+    }
+  }
+
+  private _checkImageLoaded(): void {
     let { src } = this.props;
     let { loadState } = this.state;
-    let isLoaded = (src && this._imageElement.naturalWidth > 0 && this._imageElement.naturalHeight > 0);
 
-    this._computeCoverStyle(this.props);
+    if (loadState === ImageLoadState.notLoaded || loadState === ImageLoadState.error) {
+      // testing if naturalWidth and naturalHeight are greater than zero is better than checking
+      // .complete, because .complete will also be set to true if the image breaks. However,
+      // for some browsers, SVG images do not have a naturalWidth or naturalHeight, so fall back
+      // to checking .complete for these images.
+      let isLoaded: boolean = src && (this._imageElement.naturalWidth > 0 && this._imageElement.naturalHeight > 0) ||
+        (this._imageElement.complete && Image._svgRegex.test(src));
+
+      if (isLoaded) {
+        this._computeCoverStyle(this.props);
+        this._setStateToLoadedOrErrorLoaded();
+      }
+    }
+  }
 
-    if (isLoaded && loadState !== ImageLoadState.loaded && loadState !== ImageLoadState.errorLoaded) {
-      this._events.off();
+  private _setStateToLoadedOrErrorLoaded(): void {
```

Comment:
Add comment to explain why we have errorLoaded and loaded. Or alternatively, deprecate the errorLoaded enum value because it's confusing.